CRM/Admin/Form/Preferences.php

```diff
@@ -251,6 +259,14 @@ public function postProcess() {
    * Process the form submission.
    */
   public function postProcessCommon() {
+    try {
+      $this->saveMetadataDefinedSettings($this->_params);
+      $this->filterParamsSetByMetadata($this->_params);
+    }
+    catch (CiviCRM_API3_Exception $e) {
+      CRM_Core_Session::setStatus($e->getMessage(), ts('Save Failed'), 'error');
+    }
+
     foreach ($this->_varNames as $groupName => $groupValues) {
```

Comment:
going forwards we basically migrate settings out of varNames into protected $_settings per below